### PR TITLE
[PLAT-10560] Remove empty http.flavor span attribute

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

To only add an `http.flavor` span attribute if it is a usable value.

## Testing

Updated unit tests

### CODE REVIEW

**Review by yousif-bugsnag — APPROVED** [review]

File: packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ interface ResourceTiming extends PerformanceResourceTiming {
 
 export function getHttpVersion (protocol: string) {
   switch (protocol) {
```
Comment (imjoehaines) [inline packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts:9]: Would changing the `default` instead make sense?

Comment (gingerbenw) [reply packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts:9]: probably! I guess it depends if we want to exclude all of the other [possibilities of nextHopProtocol](https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.xhtml#alpn-protocol-ids) ?

@gareththackeray any thoughts on this?

Comment (gingerbenw) [reply packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts]: The product design lists the values for `http.flavor` as `1.0 | 1.1 | 2.0 | 3.0 | SPDY | QUIC` so have gone ahead and updated it to return `undefined` for anything outside of this range.

Comment (gareththackeray) [reply packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts]: I think use watever you get back if it has a value

Comment (gingerbenw) [reply packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts]: so, just to clarify - we want to map the values to those as described in the PD (such as `h2` and h2c` to `2.0`) and anything outside of that range, we use as it is? like `xmpp` or `stun.turn`

Comment (gareththackeray) [reply packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts:9]: Yes I think so

**Review by gareththackeray — APPROVED** [review]

**Pull request merged**